go-apps/meep-metrics-engine/server/v2/metrics-engine.go

```diff
@@ -719,6 +719,105 @@ func updateSeqMetrics(log string, metric SeqMetric, params SeqQueryParams, respo
 	return metric, response
 }
 
+func mePostDataflowQuery(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	log.Debug("mePostHttpQuery")
+
+	// Retrieve sequence diagram query parameters from request body
+	var params DataflowQueryParams
+	if r.Body == nil {
+		err := errors.New("Request body is missing")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&params)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Make sure metrics store is up
+	if metricStore == nil {
+		err := errors.New("No active scenario to get metrics from")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+
+	// Make sure only type of format is specified
+	if len(params.Fields) > 1 {
+		err := errors.New("Specify only one type of format: meraid or sdorg")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Parse tags
+	tags := make(map[string]string)
+	for _, tag := range params.Tags {
+		tags[tag.Name] = tag.Value
+	}
+
+	// Get scope
+	duration := ""
+	limit := 0
+	if params.Scope != nil {
+		duration = params.Scope.Duration
+		limit = int(params.Scope.Limit)
+	}
+	// Get http metrics
+	params.Fields = append(params.Fields, met.HttpLoggerMsgType, met.HttpSrc, met.HttpDst, met.HttpLogEndpoint,
+		met.HttpBody, met.HttpMethod)
+	valuesArray, err := metricStore.GetInfluxMetric(met.HttpLogMetricName, tags, params.Fields, duration, limit)
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(valuesArray) == 0 {
+		err := errors.New("No matching metrics found")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusNoContent)
+		return
+	}
+
+	// Prepare & send response
+	var response DataflowMetrics
+	response.DataflowMetricList = &DataflowMetricList{
+		Name:    "sequence metrics",
+		Columns: append(params.Fields, "time"),
```

Comment:
Fields list should only include time & mermaid, is this the case here?

Reply:
Corrected this to only show requested fields and time